File: OrganizationService/OrganizationService.DataAccess/Models/OrganizationMember.cs
using OrganizationService.Core.Constants;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrganizationService.DataAccess.Models
{
    public class OrganizationMember : BaseTable<Guid>
    {
        public Guid UserId { get; set; }
        public ScopeType ScopeType { get; set; } // Organization, Grade, Class
        public Guid ScopeId { get; set; } // id của tổ chức, khối, lớp tương ứng
        public MemberRole Role { get; set; } // superAdmin, admin, teacher, student
        public bool IsActive { get; set; }
       
    }
}

File: org-service/src/main/java/com/codecampus/organization/entity/OrganizationMember.java
package com.codecampus.organization.entity;

import com.codecampus.constant.ScopeType;
import com.codecampus.organization.entity.audit.AuditMetadata;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
@Entity
@Table(name = "organization_members")
@SQLDelete(sql = "UPDATE organization_members SET deleted_by = 'system', deleted_at = now() WHERE id = ?")
@Where(clause = "deleted_at IS NULL")
public class OrganizationMember extends AuditMetadata {
  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  String id;

  @Column(name = "user_id", nullable = false)
  String userId; // GUID từ identity

  @Enumerated(EnumType.STRING)
  @Column(name = "scope_type", nullable = false, length = 32)
  ScopeType scopeType; // Organization | Block

  @Column(name = "scope_id", nullable = false)
  String scopeId; // id của Organization hoặc Block

  @Column(nullable = false, length = 32)
  String role; // ADMIN|TEACHER|STUDENT

  @Column(nullable = false)
  boolean isActive = true;

  @Column(nullable = false)
  boolean isPrimary = false; // primary org cho scopeType=Organization
}